sf-fx-runtime-java-runtime/src/main/java/com/salesforce/functions/jvm/runtime/commands/AbstractDetectorCommandImpl.java

```diff
@@ -94,6 +96,20 @@ public final Integer call() throws Exception {
     return handle(project, functions);
   }
 
+  private static boolean isVersionUnsupported(String salesforceApiVersion) {
+    try {
+      return parseMajor(salesforceApiVersion) < parseMajor(DEFAULT_SALESFORCE_API_VERSION);
+    } catch (Exception ignored) {
+      // if we can't parse the version we'll have to assume it's unsupported
+      return true;
+    }
+  }
+
+  private static int parseMajor(String version) {
+    String[] semverParts = version.split("\\.");
+    return Integer.parseInt(semverParts[0], 10);
+  }
+
```

Comment:
I'm not a huge fan how exceptions are used here:

- `parseMajor` does not explicitly specify the error cases via types or signature
- The try-catch catches all exceptions which is very broad

Can we use `Optional` to signal failure to parse here instead?